ha/setup/pre_disruptive_upgrade.py

```diff
@@ -123,17 +131,40 @@ def delete_resources() -> None:
         UpgradeError
     """
     try:
-        root = _get_cib_xml()
-        resources = [e.attrib["id"] for e in root.findall(".//lrm_resource")
-                     if "id" in e.attrib]
+        resources = _get_resource_list()
         Log.info(f"Going to delete following resources: {resources}")
-
         for r in resources:
-            Log.info(f"Deleting {r}")
-            SimpleCommand().run_cmd(f"pcs resource delete {r}")
+            Log.info(f"Deleting resource {r}")
+            SimpleCommand().run_cmd(f"pcs resource delete {r} --force")
+        SimpleCommand().run_cmd(PCS_CLEANUP)
+        Log.info("Wait 2 min till all resource deleted.")
+        is_resource_deleted(120)
     except Exception as err:
-        raise UpgradeError("Resource deletion failed") from err
-
+        raise UpgradeError("Resource deletion failed")
+
+def check_cluster_health():
+    """ Check cluster status and make sure cluster is healthy """
+    # Check if cluster running
+    _, _, rc = SimpleCommand().run_cmd(PCS_CLUSTER_STATUS, check_error=False)
+    if rc != 0:
+        raise UpgradeError("Cluster is not running on current node")
+    output, _, _ = SimpleCommand().run_cmd(PCS_FAILCOUNT_STATUS)
+    if "INFINITY" in output:
+        raise UpgradeError(f"Cluster is not stable, some resource are not healthy. {output}")
+
+def is_resource_deleted(timeout):
+    """ Check if pre disruptive upgrade is successful """
+    base_wait = 5
+    while timeout > 0:
+        resources = _get_resource_list()
+        Log.info(f"Waiting for {str(timeout)} to delete resources {resources}.")
+        if len(resources) == 0:
+            Log.info("All resource deleted successfully.")
+            break
+        time.sleep(base_wait)
+        timeout -= base_wait
+    if len(resources) != 0:
```

Comment:
The 'resources' is intialized in while loop. If input timeout is 0, then line #166 will fail, right?

Reply:
Updated 
now it will get resources even if timeout is 0